fairlearn/datasets/_fetch_adult.py

```diff
@@ -0,0 +1,79 @@
+# Copyright (c) Microsoft Corporation and Fairlearn contributors.
+# Licensed under the MIT License.
+
+import pathlib
+
+from sklearn.datasets import fetch_openml
+from ._constants import _DOWNLOAD_DIRECTORY_NAME
+
+
+def fetch_adult(*, cache=True, data_home=None,
+                as_frame=False, return_X_y=False):
+    """Load the UCI Adult dataset (binary classification).
+
+    Download it if necessary.
+
+    ==============   ==============
+    Samples total             48842
+    Dimensionality               14
+    Features                   real
+    Classes                       2
+    ==============   ==============
+
+    Source: https://archive.ics.uci.edu/ml/datasets/Adult
+    Ronny Kohavi and Barry Becker, "Scaling Up the Accuracy of Naive-Bayes
+    Classifiers: a Decision-Tree Hybrid", Proceedings of the Second
+    International Conference on Knowledge Discovery and Data Mining, 1996
+
+    Prediction task is to determine whether a person makes over $50,000 a
+    year.
+
+    Parameters
+    ----------
+    cache : boolean, default=True
+        Whether to cache downloaded datasets using joblib.
+
+    data_home : optional, default: None
+        Specify another download and cache folder for the datasets. By default
+        all scikit-learn data is stored in '~/fairlearn_data' subfolders.
```

Comment:
`~/.fairlearn_data` surely? I'd consider writing `$HOME` instead of `~` as well.

Reply:
Is `$HOME` cross-plat?
Because `~` works everywhere

Reply:
The exact syntax certainly isn't (although `~` doesn't work in `cmd.exe`, and that isn't quite dead yet). My concern is that `~` might get lost in the run of text. It's not critical, though.

The missing `.` OTOH..... and now I look closer, there's also `_` vs `-` in the directory name.

Reply:
`~` works in cmd.exe for me

Reply:
fixed the dir name. Must have copied that from an earlier version of fetch_boston